src/macros.rs

```diff
@@ -0,0 +1,34 @@
+#[macro_export]
```

Comment:
Some explanatory comments plz!

Reply:
Can do. What kind of explanation do you want me to write?

Reply:
A combination of: 
1. what each line does
2. a very quick explainer of the thing as a whole... how it functions and what it enables

Reply:
Especially the ParamStruct + Serialize/Deserialize bits

Reply:
Okay, I'll write a doc comment about the usage